### PR TITLE
component incompatibility mapping

### PR DESCRIPTION
**Pull request by Raulp8**
- printing base incompatibility report for current documented mappings

### CODE REVIEW

File: incompatibility/incompatibility-report/incomp-scanning.go [hunk 1]
```diff
@@ -34,7 +37,26 @@ func CreateIncompReport(env *plugins.Environment, reportType Report) {
...
+		if model.TypeUrl != "openapi.v3.Document" {
+			continue
+		}
+		openAPIdocument := &openapiv3.Document{}
+		err := proto.Unmarshal(model.Value, openAPIdocument)
+		if err != nil {
+			continue
+		}
+		printIncompatibilityReport(ScanIncompatibilities(openAPIdocument))
+
```
Comment (AlanPeters) [inline incompatibility/incompatibility-report/incomp-scanning.go:50]: nit: There is some extra space here

File: incompatibility/incompatibility-report/reporter.go [hunk 2]
```diff
@@ -194,7 +248,118 @@ func schemaSearch(schema *openapiv3.Schema, keys []string) []*Incompatibility {
...
+}
+
+//  headerSearch scans for incompatibilities in a header object
+func headerSearch(headerName string, header *openapiv3.Header, keys []string) []*Incompatibility {
+	var incompatibilities []*Incompatibility
+	if header == nil {
+		return incompatibilities
+	}
+	paramEquiv := header2Paramter(headerName, header)
+	incompatibilities = append(incompatibilities,
```
Comment (AlanPeters) [inline incompatibility/incompatibility-report/reporter.go:327]: Nice work here. I am sure it was tempting to inline this but adding the headerSearch as its own function does a good job of communicating the relation between headers and parameters

**Review by AlanPeters — APPROVED** [review]
Couple of minor nits but looks good to me

**Review by skeid21 — APPROVED** [review]

File: incompatibility/incompatibility-report/incomp-scanning.go [hunk 3]
```diff
@@ -34,7 +37,26 @@ func CreateIncompReport(env *plugins.Environment, reportType Report) {
 
 	// If indicated by report type associate incompatibilities with line
 	// references, etc. and generate an ID_REPORT
+	for _, model := range env.Request.Models {
+		if model.TypeUrl != "openapi.v3.Document" {
+			continue
```
[the remark this answers is not in the document]
Comment (Raulp8) [reply incompatibility/incompatibility-report/incomp-scanning.go:42]: user can't indicate which model is passed to plugin so it wouldn't be of interest imo to indicate we're sklipping over a model. However, I'm going to include an error if we can support any of the models in the request

File: incompatibility/incompatibility-report/incomp-scanning.go [hunk 4]
```diff
@@ -34,7 +37,26 @@ func CreateIncompReport(env *plugins.Environment, reportType Report) {
...
+		}
+		openAPIdocument := &openapiv3.Document{}
+		err := proto.Unmarshal(model.Value, openAPIdocument)
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (Raulp8) [reply incompatibility/incompatibility-report/incomp-scanning.go:46]: now using env.respondandexitiferror(err)

File: incompatibility/incompatibility-report/reporter.go [hunk 5]
```diff
@@ -88,7 +89,60 @@ func PathsSearch(doc *openapiv3.Document) []*Incompatibility {
...
+
+		for ind, paramOrRef := range pathValue.Parameters {
+			incompatibilities = append(incompatibilities,
+				parametersSearch(paramOrRef.GetParameter(), addKeyPath(pathKey, "parameters", strconv.Itoa(ind)))...)
+		}
+	}
+	return incompatibilities
+}
+
+func ComponentsSearch(doc *openapiv3.Document) []*Incompatibility {
```
[the remark this answers is not in the document]
Comment (Raulp8) [reply incompatibility/incompatibility-report/reporter.go:101]: resolved

**Pull request merged**